lib/Sema/TypeChecker.h

```diff
@@ -2083,7 +2083,8 @@ class TypeChecker final : public LazyResolver {
                              unsigned maxResults = 4);
 
   void noteTypoCorrection(DeclName name, DeclNameLoc nameLoc,
-                          const LookupResult::Result &suggestion);
+                          const LookupResult::Result &suggestion,
+                          FunctionRefKind refKind = FunctionRefKind::Unapplied);
```

Comment:
I'm still not 100% sure this is the right type for this.  For names that aren't function references that go through typo correction this is kind of nonsensical.